app/javascript/components/App.jsx

```diff
@@ -137,14 +147,11 @@ class App extends Component {
             signup: this.signup,
             continueAsGuest: this.continueAsGuest
           }}>
+            {this.state.loading ? loader : <span></span>}
             <ResponsiveContainer                      
             >
-              {this.state.loading
-              ? (
-                <Dimmer active inverted>
-                  <Loader inverted>Loading</Loader>
-                </Dimmer>
-              )
+              {this.state.initialLoading
```

Comment:
this one is for the app not to destroy all the other components whenever login/logout/signup is called.